### PR TITLE
Genesis pods [WIP]

### PR DESCRIPTION
**Pull request by 4dplanner**
:cl:
add: Exotic seeds crates and the seed vault now contain genesis pods, which produce new sentient life!
/:cl:

Basic idea is that when the plant is harvestable a ghost can click it to become a new podperson. Very WIP, but I wanted to put up a prototype for discussion.

Todo:

- [ ] sprites
- [ ] Make potency actually matter?
- [x] make a way to get it 
- [x] limit yield in a similar way to replica pods
- [x] make ghost spawn actually harvest the plant
- [x] greater name variety
- [ ] flavour text?

### CODE REVIEW

**Comment by 4dplanner** [comment]
I'd appreciate suggestions on what to do with potency - if it just makes the podperson produced stronger in some way it'd feel pretty crappy if you were created by a bad botanist and had your potential curtailed for the rest of your life. 

One possibility is to base the podman's light healing factor off of the potency, so it's not something vital to day to day life but it is something nice to have.

**Comment by zxaber** [comment]
To be clear, a ghost cannot take a pod that has blood added to it, right? That is, the botanist doesn't have to "beat" a ghost to clicking it in order to get the person they're trying to revive back, I assume.

For potency, maybe have a dice roll to see if the plant can produce an additional pod after the prior one was taken, with a higher chance based on the potency? Perhaps with enough potency, you can have a single plant pump out three or four pods before finally dying?

Also, the pod people, unlike xeno friends, are unbound to anyone, right?

**Comment by 4dplanner** [comment]
Genesis pods are a different species to replica pods, so that won't be an issue. Replica pods are unchanged.

 I'd certainly intended them to be unbound, given that botanists are powerful enough already. 

As to your suggestion, that might be possible, but it'd lead to an overlap with yield in terms of stat effect (yield leads to more seeds if you manually harvest)

**Comment by kingoftheautism** [comment]
Maybe leave potency as reversed equalent of clone damage? Less potency - more clone damage, more chances that you will miss one or more limb/organ, etc. It probably should start actually affecting genesis pod persons if there is <40/50 potency. I'm not really fan of that idea tho, since it's adding less challenge to botanists, who wants to make cloning 2.0. It can possibly became not challenge but rather frustration too.

Basically, I'm gonna let people comment with their ideas, what potency should do

**Comment by GuyonBroadway** [comment]
Dumb idea but what if potency was tied to the mob type it spawned?

91-100 = podman
90-60 = plant based monkey
59-30 = plant based dog (a border cauli-flower)
<30 = plant based mouse



Look I just wrote this to make a a terrible border-cauli joke. At the same time it does sound amusing that botanists could manage thier plant potency for unsusual effects and it be different from "go 100 or go home"

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 1]
```diff
@@ -16,7 +16,7 @@
 	for contact from your creators. Estimated time of last contact: Deployment, 5x10^3 millennia ago.</b>"
 	assignedrole = "Lifebringer"
 
-/obj/effect/mob_spawn/human/seed_vault/special(mob/living/new_spawn)
+/proc/name_podperson(mob/living/new_spawn)
 	var/plant_name = pick("Tomato", "Potato", "Broccoli", "Carrot", "Ambrosia", "Pumpkin", "Ivy", "Kudzu", "Banana", "Moss", "Flower", "Bloom", "Root", "Bark", "Glowshroom", "Petal", "Leaf", \
 	"Venus", "Sprout","Cocoa", "Strawberry", "Citrus", "Oak", "Cactus", "Pepper", "Juniper")
 	new_spawn.fully_replace_character_name(null,plant_name)
```
Comment (XDTM) [inline code/game/objects/structures/ghost_role_spawners.dm]: Put this in the species name generation proc

**Comment by XDTM** [comment]
By the way, if hydroponics growing people from scratch is ok then robotics should be able to create androids with cybernetic brains

**Comment by GuyonBroadway** [comment]
Honestly, as long as it removes antag status, comes with an exploitable downside and can't be done en-masse at roundstart ut sounds ok to me.

It's like xenobio golems except they die to plant-b-gone/the ion gun.


Consider robotics could mass fabricate borgs from posibrains anywho then weaker humans isn't that bad.


Heck, just being able to cram a posibrain into a debrained human would be  neat enough but emps cause brain damage.

**Comment by kingoftheautism** [comment]
I already had idea about some kind of adapter for positronic brains, so you can put them into human bodies, maybe it will work and robotics will make androids, maybe it wont because people wont like it

**Comment by GuyonBroadway** [comment]
I see getting players back into the round as a good thing. Especially if it's stuff like being borged or golems as opposed to autocloners. 

Autocloners let a dead muderboner live to bone again. Becoming a new mob eliminates the antag but keeps the player in the round.

**Comment by MACIEKBAKI** [comment]
I like the idea of more ghost roles if the station's botanists decide to actually _do something useful_

**Comment by GuyonBroadway** [comment]
These aren't enslaved to the botanist who grew em so it's not like its a free botany army.

And setting up a cult rune in the cold light of botany is a good way for every greenthumb to get a baton up the bum.

**Comment by 4dplanner** [comment]
> near-roundstart golems lol...

The reason I thought this was OK was because without loyalty they aren't actually at all useful for powergaming.

**Comment by GuyonBroadway** [comment]
You COULD use it to bulk out a revs army but at the end of the day it's not gonna be much better than revgolems.

**Comment by 4dplanner** [comment]
> You COULD use it to bulk out a revs army but at the end of the day it's not gonna be much better than revgolems.

It'd actually be pretty overpowered as revs, given how quickly botany can get mutatations. Not sure how to solve this.

**Comment by Ziiro** [comment]
Remove their eyes and let them see without eyes? So they can't be flashed. Could still be culted though. Might be outside of the scope of this PR, though.

**Comment by Ratvar** [comment]
If they can't be culted - they can be freely used to fight cult. There's no alternative and no solution untill you make them ignore antagonists altogether, and TG doesn't work like that/it sucks.

**Comment by 4dplanner** [comment]
>add some oxygen generating plant 

I like this idea. 

So far I'm planning on just hoping the slow growth of the plant is enough to make this not a problem, but we'll see.

**Comment by davethwave** [comment]
Another interesting plant idea is one that could scrub toxins from the air. Like plasma and stuff.

**Comment by Barhandar** [comment]
Add plants for all the gases, let engineering irradiate them to fuck and back to generate fusion gases instead, watch as engineering fills entire station with plox and stimulum plants.

**Comment by FantasticFwoosh** [comment]
> 
> 
> Dumb idea but what if potency was tied to the mob type it spawned?
> 
> 91-100 = podman
> 90-60 = plant based monkey
> 59-30 = plant based dog (a border cauli-flower)
> <30 = plant based mouse

Perfectly doable, i did something like this for my gatfruit PR a long time ago.  #26751 But i dont really believe in this being a mutatable plant rather than something you find as a reward with unsustainable base statistics because replica pods are gamestart, and easily mutated for more from cabbages by powergamers.

Codewise Its just ``addjuice`` if you dive in the code.

**Comment by 4dplanner** [comment]
It's not powerful (in most cases), so it's not a problem if it's relatively easy to get. Creating more ghostroles is the point of this PR.

**Comment by MimicFaux** [comment]
Idea: Genesis pod requires soil plot to grow (the larger plant needs more room) and a hydrotray is too small which can only be used to grow harvestable seeds.
Makes growing one a more hands-on experience and careful tending to. If the botanist uses Gaia on the soil, well, they've already hit their stride.

Requires additional setup, harder to mass produce, and has a tidy IC reason for the roadblock.
On top of the potency thing, because I want my Border Cauli-flower damnit.

**Comment by XDTM** [comment]
Seed vault is a good idea though, would be the same mechanic as golems with the golem creation disk, where they have an easier method of generating more people than common station methods.

**Comment by 4dplanner** [comment]
Border Cauli is a brilliant pun, but I have no artistic talent so it's not possible. If anyone wants to give me some sprites so I'm not just colour shifting the replica pods be my guest ;)

As to it being lavaland only, I'd originally specifically intended it to be a way for dead people to play a part on the station. That's one of the good things about (on station) golems - you can dress a golem up as a doctor and you have a new doctor. This was intended to be an easier path to that balanced by the inherent weaknesses of podpeople and the lack of antag potential. 

I'm not totally set on this point, so if the consensus is that they should be in the seed vault vendor then I'll change it, I just feel there's also space for more ways to get people into the round in a straightforward, crewmember on a space station way. 

Alternatively, if people understood that it was totally fine to spawn in as podpeople just to try and get to the station, that could make for a cool dynamic. The issue is that currently roles like (I forget the name) where you spawn in a broken shelter next to a space suit are completely ignored, despite the potential for making your way to mining and getting into the round as a new character.

**Comment by 4dplanner** [comment]
Making them require a careful, hands on experience would kill them. Who would put that much effort into something that doesn't even benefit them when they could make more exploding tomatoes?

**Comment by 4dplanner** [comment]
Also I haven't played much of them, so do seed vaulters have a way to heal cloneloss?

**Comment by 4dplanner** [comment]
Note that it's a random spawn in seed vault, so they won't always have them.

**Comment by Ratvar** [comment]
You heal cloneloss by reviving in podplant

**Comment by 4dplanner** [comment]
Will probably add the low potency = cloneloss idea, then

**Comment by Ratvar** [comment]
but cloneloss sucks tho.

**Comment by JStheguy** [comment]
Maybe just have low potency start the plant person with regular old brute damage or something? That way there is still the initial issue of starting in sub-optimal condition, but it can be gotten over with medication or just waiting around a while for your body to get itself into peak shape. Maybe make it toxic damage instead if you want to avoid getting over it by just sitting on your increasingly fat ass soaking up sweet, sweet light.

**Comment by optimumtact** [comment]
there's no ghosts at roundstart who want to play anyway you 4heads

**Comment by davethwave** [comment]
But your only considering roundstart though... There is always that person who says yes to the defective clone prompt when it comes up.

**Comment by habader** [comment]
I was thinking that a low potency could spawn mobs rather than being producing a body. Something like a killer tomato, venus human trap, a few spiderlings or bees

**Comment by plapatin** [comment]
diona nymphs when

**Comment by ExcessiveUseOfCobblestone** [comment]
Any updates on this?

**Comment by 4dplanner** [comment]
The person who said they were going to do sprites for this didn't come through, so it's on hold for the momentmoment. I'll close it until I'm done with IRL stuff and cam try making

**Comment by 4dplanner** [comment]
The person who said they were going to do sprites for this didn't come through, so it's on hold for the momentmoment. I'll close it until I'm done with IRL stuff and cam try making

**Pull request closed without merging**